src/bootstrap/builder.rs

```diff
@@ -1271,7 +1270,11 @@ impl<'a> Builder<'a> {
         }
 
         // For `cargo doc` invocations, make rustdoc print the Rust version into the docs
-        cargo.env("RUSTDOC_CRATE_VERSION", self.rust_version());
+        // This replaces spaces with a Unicode en-space because RUSTDOCFLAGS
+        // does not support arguments with regular spaces. Hopefully someday
+        // Cargo will have space support.
+        let rust_version = self.rust_version().replace(' ', "\u{2002}");
+        rustdocflags.arg("--crate-version").arg(&rust_version);
```

Comment:
This might be a bit out there, but should work more uniformly across platforms I think. If cargo is splitting by regular ASCII space only, we could use a new line here.

That said, thinking some more about this it seems really error prone - anyone can replace that splitting with split_whitespace which would break us here. But I feel like that's not too big a deal, we can fix it as needed. Would you be up for adding a Cargo test as well so that the (potential) breakage is detected earlier?

Reply:
Ah, I like newline better.

I'll put it on my todo list to add a regression test in cargo.